src/Product.php

```diff
@@ -1038,11 +1053,9 @@ protected function processTexts(array $data)
             $this->setField('name', $this->getFromArray($texts, 'name' . $this->productNameFieldId))
                 ->setField('summary', $this->getFromArray($texts, 'shortDescription'))
                 ->setField('description', $this->getFromArray($texts, 'description'))
-                ->setField('url', $this->getProductFullUrl(
-                    $this->getFromArray($texts, 'urlPath'),
-                    $this->getFromArray($data, 'mainVariationId')
-                ))
                 ->setField('keywords', $this->getFromArray($texts, 'keywords'));
+
+            $this->path = $this->setPath($this->getFromArray($texts, 'urlPath'));
```

Comment:
The setter returns an object, so it shouldn't be set to `$this->path`

Reply:
Good that there are tests :p. Already fixed it ^^.